File: Guess.rb
rando = rand(1..100)

max_attempts = 10

puts "Greetings guest! If you can correctly guess the number from 1-100, You will receieve a fabulous prize!"

1.upto(max_attempts) do |attempt|
  print "Attempt #{attempt} of #{max_attempts}: "
  guess = gets.to_i

  if guess == rando
    puts "Ayyyy nice job! You earned a prize!"
    break
  elsif (guess - rando).abs > 20
    puts "OhNAH you are FREEZING!"
  elsif (guess - rando).abs > 10
    puts "Feelin a bit cold there buddy..."
  elsif (guess - rando).abs > 5
    puts "Now we're warming up!"
  else
    puts "Ooooh you are so so warm. ALMOST THERE!"
  end

  if attempt == max_attempts
    puts "Darn.. You're out of attempts. The correct number was #{rando}."
  end
end

File: Guess.cpp
#include <iostream>
#include <cstdlib>
using namespace std;

int main() {
    int guess;
    int maxattempts = 10;
    int rando;

    rando = (rand() % 100) + 1;

    cout << "Greetings guest! Guess a number from 1-100 for a prize!" << endl;

    for(int attempts = 1; attempts <= maxattempts; attempts++) {
        cout << "Attempt " << attempts << " of " << maxattempts << ": ";
        cin >> guess;

        if (guess == rando) {
            cout << "Great job! Come claim your prize!" << endl;
            break;
        }
        else if (abs(guess - rando) > 20) {
            cout << "Ooooh you are FREEZING!" << endl;
        }
        else if (abs(guess - rando) > 10) {
            cout << "Hmmm... getting cold there..." << endl;
        }
        else if (abs(guess - rando) > 5) {
            cout << "You're getting warmer!" << endl;
        }
        else {
            cout << "Ooooh you are so so warm. ALMOST THERE!" << endl;
        }

        if (attempts == maxattempts) {
            cout << "Uh oh! You fumbled! Well the correct number actually was " << rando << "." << endl;
        }
    }

    return 0;
}
